python/truelayer_signing/utils.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import annotations
+
+# std imports
+import base64
+import json
+from enum import Enum
+from typing import Dict, Mapping, MutableMapping, Optional, Tuple, TypeVar
+
+T = TypeVar('T', bound='TlJwsBase')
+
+
+class HttpMethod(str, Enum):
+    POST = "POST"
+    GET = "GET"
+    PATCH = "PATCH"
+    PUT = "PUT"
+    DELETE = "DELETE"
+
+
+class TlJwsBase:
+    pkey: str
+    http_method: HttpMethod
+    path: str
+    headers: MutableMapping[str, str]
```

Comment:
Same caveat as in `sign.py` applies here, ordering matters and `dict`s have an implicit and unspecified order that changes from instance to instance